drivers/ethernet/eth_sam_gmac.c

```diff
@@ -75,49 +75,243 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 /* RX descriptors list */
 static struct gmac_desc rx_desc_que0[MAIN_QUEUE_RX_DESC_COUNT]
 	__nocache __aligned(GMAC_DESC_ALIGNMENT);
+#if GMAC_QUEUE_NUM > 1
 static struct gmac_desc rx_desc_que1[PRIORITY_QUEUE1_RX_DESC_COUNT]
 	__nocache __aligned(GMAC_DESC_ALIGNMENT);
 static struct gmac_desc rx_desc_que2[PRIORITY_QUEUE2_RX_DESC_COUNT]
 	__nocache __aligned(GMAC_DESC_ALIGNMENT);
+#endif
+#if GMAC_QUEUE_NUM > 3
+static struct gmac_desc rx_desc_que3[PRIORITY_QUEUE3_RX_DESC_COUNT]
+	__nocache __aligned(GMAC_DESC_ALIGNMENT);
```

Comment:
I believe that is being addressed in #23031 (in particular, https://github.com/zephyrproject-rtos/zephyr/pull/23031/commits/5e4074aae800e5f277d7a205382aef3d2e7fbecb).

Since there is currently no SAM 4E GMAC support (without #23031), this should not be an issue as far as this PR (bugfix) is concerned.

Reply:
I decided to move any non priority queue related change to #23031